applications/fishing-map/src/features/resources/resources.slice.ts

```diff
@@ -26,14 +30,24 @@ const initialState: ResourcesState = {}
 
 export const fetchResourceThunk = createAsyncThunk(
   'resources/fetch',
-  async (resource: ResourceQuery) => {
+  async (resource: ResourceQuery, { getState }) => {
+    const state = getState() as RootState
+    const guestUser = isGuestUser(state)
+    const { thinning } = selectDebugOptions(state)
     const isTrackResource = resource.datasetType === DatasetTypes.Tracks
     const responseType =
       isTrackResource &&
       resource.datasetConfig.query?.some((q) => q.id === 'binary' && q.value === true)
         ? 'vessel'
         : 'json'
-    const data = await GFWAPI.fetch(resource.url, { responseType }).then((data) => {
+    let url = resource.url
+    if (isTrackResource && thinning) {
```

Comment:
See my dataviews comment - the problem here is that we introduce track-specific logic in a generic resource thunk

Reply:
Agreed on this should be dataview responsibility instead of doing it in the thunk, it has been updated
Also there is a [new enum](https://github.com/GlobalFishingWatch/frontend/pull/375/commits/d99bf5f30bceed021ac990a0de8b4e992189be73#diff-d1dbd13d07dbb0113e847e8262d6284320eee1a5c5fb1ce2d55a2a31f21e000dR17) to better use of the API dataset endpoints